memcached_stat.py

```diff
@@ -69,44 +65,41 @@ def make_synthetic_stats(stats):
     # create the slab memory size (chunk_size*total_chunks for each slabs)
     chunk_sizes={}
     total_chunks={}
-    for s in stats['slabs']:
-        if ':' not in s[0]:
+    for k,v in stats['slabs'].iteritems():
+        if ':' not in k:
             continue
-        slabid, key = s[0].split(':')
+        slabid, key = k.split(':')
         if key == 'chunk_size':
-            chunk_sizes[slabid] = s[1]
+            chunk_sizes[slabid] = int(v)
         elif key == 'total_chunks':
-            total_chunks[slabid] = s[1]
+            total_chunks[slabid] = int(v)
     # prepare the intersection of keys in both maps
     slabids = set(chunk_sizes.keys()) & set(total_chunks.keys())
     log_verbose("preparing synthetic metrics for %d slabs"%len(slabids))
     for slabid in slabids:
-        stats['slabs'].append(["synth:%s:slab_size"%slabid, int(chunk_sizes[slabid])*int(total_chunks[slabid])])
-        log_verbose("memcached_stats plugin: craeted a synthetic slab_size metric for slabid %s"%slabid)
+        stats['slabs']["synth:%s:slab_size"%slabid] = chunk_sizes[slabid] * total_chunks[slabid]
+        log_verbose("memcached_stats plugin: created a synthetic slab_size metric for slabid %s"%slabid)
 
 def read_callback():
     for conf in CONFIGS:
         stats=get_memcached_stats(conf)
         make_synthetic_stats(stats)
         for verb,stat in stats.iteritems():
-            stat = filter(lambda s: s[0].split(':')[-1] in conf['filtered_stat_types'], stat)
+            # filtering stats that are mentionned in the configuration
+            fstat = dict((k, v) for k, v in stat.iteritems() if k.split(':')[-1] in conf['filtered_stat_types'])
             # fix keys by prepending the verb if it doesn't exist
-            stat=map(lambda s: [verb+':'+s[0],s[1]] if not s[0].startswith(verb) else s, stat)
-            for s in stat:
-                log_verbose("Having a stat[%s]=%s"%(s[0], s[1]))
-                dispatch_value(s[0],conf['filtered_stat_types'][s[0].split(':')[-1]],s[1], conf['instance'])
+            ffstat = dict((verb+':'+k if not k.startswith(verb) else k, v) for k, v in fstat.iteritems())
```

Comment:
I tried and found the result unreadable... since we're talking of a few 10's of stat lines every minute or so, I thought it's ok, However if you find a readable one-liner for all this, I would be glad to use it, I also don't like iterating twice!